### ISSUE TITLE
Ability to create pages dynamically?

### ISSUE THREAD

**Issue by gkostyanikov** [spf13/hugo#140]
Using this feature provided by Middleman http://middlemanapp.com/dynamic-pages/ quite often. So maybe somebody will point me to the machinery of doing similar stuff in Hugo or consider adding this feature in future releases?

----
Edit: New link at https://middlemanapp.com/basics/dynamic-pages/

**Comment by anthonyfok** [comment]
Update: @SchumacherFM has submitted a wonderful Pull Request #748: "Feature: GetJson and GetJson in short codes or other layout files" as a solution to this issue.

**Comment by bep** [comment]
We now have both Data Files and Data driven content -- this is as dynamic as one would get for a static site.

**Comment by lukasz-kaniowski** [comment]
Hi guys, 
I'm trying to generate a list of book description pages from a json that contains a list of them, i.e:
```
[
  { "title", "book A", "description": "description A"},
  { "title", "book B", "description": "description B"},
  { "title", "book C", "description": "description C"}
] 
```
I would like to get 3 pages with details of given book (generated using same layout). Is that possible?

**Comment by bep** [comment]
@lukasz-kaniowski not with Hugo alone ...

**Comment by lukasz-kaniowski** [comment]
@bep any recommendations? I'm creating this json file using node scrapper so I can change the sctructure, i.e. devide it into single json file per book. will this help?

**Comment by bep** [comment]
@lukasz-kaniowski no, you would need to create a script outside Hugo to generate the content pages.

**Comment by lukasz-kaniowski** [comment]
@bep bummer. thanks for your help.

**Comment by SchumacherFM** [comment]
Here is your solution:

http://cyrillschumacher.com/2015/02/09/virtual-pages-with-gohugo.io/

I never made a PR with this feature. It needs more love => Refactoring and hooking into other internal hugo stuff ...

**Comment by lukasz-kaniowski** [comment]
@SchumacherFM nice one. i will have a go with it.

**Comment by SchumacherFM** [comment]
Good luck.

**Comment by kpym** [comment]
Why is this issue close ? Is it because there is no hope to see pages generated for each record in a data set ? Or is it because this was implemented in some way since 2015 ?

If this was not done already I want to advocate for this feature.

**Comment by DanailMinchev** [comment]
@bep @spf13 
Could you confirm for clarity please:

There is no way to generate dynamic page currently in Hugo out of the box and it will not be implemented?

Details:
Perfect example of dynamic page will be https://middlemanapp.com/basics/dynamic-pages/ as pointed by @gkostyanikov 
There is one workaround mentioned here: https://cyrillschumacher.com/2015/02/09/virtual-pages-with-gohugo.io/ by @SchumacherFM (thanks @SchumacherFM ), but this solution does not include adding any dynamic data to the `Front Matter` ( @SchumacherFM please correct me if I am wrong).
Basically, what I want is to generate dynamic pages using data from remote JSON API or any other datasource / I can generate static json files as well, but to be able fully control generated pages like **url**, **title**, **taxonomies** or **any data parameters** which should be rendered as part of the content.

Please it will be very useful to clarify that.

THANKS

**Comment by kidsil** [comment]
So this thing has been bugging me for a while now (especially as I wanted to make it easier for contributors on my website to add content without manually creating folders & files).
So I wrote a temporary solution: **[kidsil/hugo-data-to-pages](https://github.com/kidsil/hugo-data-to-pages)**

It's a JS wrapper that generates pages from data (and it cleans after itself!).
I'm currently using this to generate pages from YAML files and it seems to be working perfectly. 
Would definitely appreciate some feedback.

Cheers
